core/services/job/orm.go

```diff
@@ -237,25 +237,58 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 					return errors.Wrapf(ErrNoSuchKeyBundle, "%v", jb.OCR2OracleSpec.OCRKeyBundleID)
 				}
 			}
+
 			if jb.OCR2OracleSpec.TransmitterID.Valid {
 				switch jb.OCR2OracleSpec.Relay {
 				case relay.EVM:
 					_, err := o.keyStore.Eth().Get(jb.OCR2OracleSpec.TransmitterID.String)
 					if err != nil {
 						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
 					}
+
+					newChainID, err := EVMChainIDForJobSpec(jb.OCR2OracleSpec)
+					if err != nil {
+						return err
+					}
+
+					var spec OCR2OracleSpec
+					chainIdPath := []string{"chainID"}
+					err = tx.Get(&spec, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2 AND relay_config #>> $3 = $4 LIMIT 1`,
+						relay.EVM, jb.OCR2OracleSpec.ContractID, chainIdPath, fmt.Sprintf("%v", newChainID),
+					)
+
+					if !errors.Is(err, sql.ErrNoRows) {
+						if err != nil {
+							return errors.Wrapf(err, "db read error while validating contract_id")
+						}
+						chainID, err := EVMChainIDForJobSpec(&spec)
+						if err != nil {
+							return errors.Wrapf(err, "internal error while validating contract_id")
+						}
+						return errors.Errorf("Job ID %v already exists for chain ID %v with contract address %v",
+							jb.ID, chainID, jb.OCROracleSpec.ContractAddress)
```

Comment:
Can we just use the other chain id var?
```suggestion
						return errors.Errorf("Job ID %v already exists for chain ID %v with contract address %v",
							jb.ID, newChainID, jb.OCROracleSpec.ContractAddress)
```